### ISSUE TITLE
Http has been deprecated, change to HttpClient

### ISSUE THREAD

**Issue by tonywr71** [aspnet/JavaScriptServices#1211]
I'm researching sending validation tokens and authorisation through angular 4 at the moment, and find I need to use the HttpClient service (first available in Angular 4.3), not the Http service, and, in fact, according to Maximus the Http service has been deprecated. See Maximus comment in the thread [here](https://stackoverflow.com/questions/45129790/difference-between-http-and-httpclient-in-angular-4)

The official documentation on the new improved HttpClient may be found [here](https://angular.io/guide/http).

**Comment by SteveSandersonMS** [comment]
Thanks for the pointer. We'll move over to the newer API.

**Comment by vasyliv** [comment]
[Tonywr71 ](https://github.com/tonywr71) How about load data at the server side? Does HttpClient work at the server side?

**Comment by tonywr71** [comment]
No, Httpclient is client side only. You would need to do the server side
part yourself. I still haven't completed this bit myself.

The anti forgery token is split into two parts, one is the cookie that is
http only and is sent in the header. The second is a token that, in cshtml,
it's put in a hidden input field within a form on the page.

What normally happens is that when you post that back to the server, the
server validates the cookie and token.

Within angular, it expects that token to have a particular name (but can be
changed if need be) and it also needs it to be javascript accessible, not
http only.

If angular sees the name it expecting, it will automatically send that with
each request.

It does this via the newly introduced HttpClient service, not the old Http
service.

On 25 Aug 2017 7:53 AM, "Mihail" <notifications@github.com> wrote:

Tonywr71 <https://github.com/tonywr71> How about load data at the server
side? Does HttpClient work at the server side?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/aspnet/JavaScriptServices/issues/1211#issuecomment-324767767>,
or mute the thread

**Comment by tonywr71** [comment]
I should add that it would be awesome if the template handled cross site forgery out of the box!

**Comment by jrmcdona** [comment]
@tonywr71 are you referring to the Angular universal flow where server renders first?   So that part isn't built in?

**Comment by tonywr71** [comment]
I'm sure when they upgraded to HttpClient they would have ensured that it also worked with universal. If something breaks it would be because they found a different, probably better way to do something and code should be modified to support new way going forward. 

Oh, and for those wondering how to set up the server side anti-forgery, there is a great article [here](http://www.dotnetcurry.com/aspnet/1343/aspnet-core-csrf-antiforgery-token) that shows how to do it. This will work on any http to webapi requests within the single page (in my case the angular app), but not on requests across .cshtml pages (you will need to do traditional cross site forgery checks in that case. I do this for Individual Security/Asp.Net Membership, which I have working - another feature I feel should be provided out-of-the-box in the template, but I understand the guys have time pressures.)

**Comment by Singularity222** [comment]
So, it seems HttpClient does work on the server, or maybe I'm misunderstanding something. I created a new Angular project using dotnet new. Upgraded the Angular packages to 4.4.5 from 4.2.5.
 I updated Http to HttpClient in the fetchdata.component.ts file, then ran both webpack config files. Started the app up and force refreshed on the fetch data page. I can see in the initial response, there is html coming from the server for all the rows in the table. I was considering upgrading to HttpClient from Http in another project and was testing it out before I went ahead with it. Am I missing something? Seems like server side rendering is working with the new client.
